roscala/src/main/scala/coop/rchain/rosette/VirtualMachine.scala

```diff
@@ -605,8 +635,13 @@ object VirtualMachine {
     for {
       _ <- modify[VMState](_.set(_ >> 'ctxt >> 'nargs)(op.nargs))
 
-      prim   = Prim.nthPrim(op.primNum)
-      result <- runPrim(op.unwind, prim)
+      prim          = Prim.nthPrim(op.primNum)
+      runPrimResult <- runPrim(op.unwind, prim)
+
+      (result, optContinuation) = runPrimResult
+
+      // Schedule continuation if there is one
+      _ <- optContinuation.map(schedule).getOrElse(pure(()))
```

Comment:
Quite repeated pattern, could it be factored out?

Reply:
Done.